packages/audio-block/src/AudioBlock.tsx

```diff
@@ -120,6 +83,26 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
         }
     };
 
+    const onTitleChange = (value: string) => {
+        if (value === blockSettings.title) {
+            setIsApiRequestPending(false);
+        } else {
+            setBlockSettings({ title: value }).finally(() => {
+                setIsApiRequestPending(false);
+            });
+        }
+    };
+
+    const onDescriptionChange = (value: string) => {
+        if (value === blockSettings.description) {
+            setIsApiRequestPending(false);
```

Comment:
small edgecase that might be worth considering: Title is edited, then user clicks in the description and then refreshes the page. This case the data might be lost because the onDescriptionChange function is called on click as well. That would set the isApiRequestPending to false even if the previous request is still pending.

Solution: Use separate variables for all settings fields. Or an object.

What do you think?